java/code/src/com/redhat/rhn/frontend/action/user/BaseUserSetupAction.java

```diff
@@ -68,14 +67,31 @@ public void setCurrentLocale(RequestContext ctx, User user) {
         }
     }
 
+    /**
+     * Sets the locale to be used for the documentation to the provided request context
+     * @param ctx RequestContext
+     * @param user User
+     */
+    public void setDocsLocale(RequestContext ctx, User user) {
+        String docsLocale = user.getPreferredDocsLocale();
+
+        // If user has locale set, then just use that
+        if (docsLocale != null) {
+            ctx.getRequest().setAttribute("currentDocsLocale", docsLocale);
+        }
+        else {
+            ctx.getRequest().setAttribute("currentDocsLocale", "default");
+        }
```

Comment:
Maybe using `Optionals` you could get rid of this `if/else` condition and do everything in one line. Just an improvement suggestion though.

Reply:
I used a different approach since I couldn't fit the optional thing in one line: using `ifPresentOrElse()`.
Just have another look